src/react-integration/NetworkErrorBoundary.tsx

```diff
@@ -1,27 +1,31 @@
 import React from 'react';
 
+export interface ErrorWithStatus extends Error {
+  status: number;
+}
+
 interface Props {
   children: React.ReactNode;
-  fallbackComponent: React.ComponentType<{ error: Error }>;
+  fallbackComponent: React.ComponentType<{ error: ErrorWithStatus }>;
 }
 interface State {
-  error?: Error;
+  error?: ErrorWithStatus;
 }
 export default class NetworkErrorBoundary extends React.Component<
   Props,
   State
 > {
   static defaultProps = {
-    fallbackComponent: ({ error }: { error: any }) => (
+    fallbackComponent: ({ error }: { error: ErrorWithStatus }) => (
       <div>
         {error.status} {error.response && error.response.statusText}
       </div>
     ),
   };
 
-  static getDerivedStateFromError(error: any) {
+  static getDerivedStateFromError(error: ErrorWithStatus | any) {
```

Comment:
`| any` might as well just be any. Why did you keep it around?

Reply:
because we know it can be ErrorWithStatus and that's the type that interests us. if you prefer I can change it back to any ?

Reply:
ah right so you know what you're setting.

Might I suggest a user-defined type-guard then:

```typescript
function isNetworkError(error: NetworkError | any): error is NetworkError {
    return !!(error as NetworkError).status;
}
```

then

```typescript
if (isNetworkError(error)) {
  return { error };  // after if block error will be typed as NetworkError
}
```

(read more here: https://www.typescriptlang.org/docs/handbook/advanced-types.html)